src/nginx/t/cloud_trace.t

```diff
@@ -134,27 +143,39 @@ is($json_obj->{traces}->[0]->{spans}->[0]->{name},
 is($json_obj->{traces}->[0]->{spans}->[0]->{kind}, 'RPC_SERVER', 'Trace span kind is RPC_SERVER');
 is($json_obj->{traces}->[0]->{spans}->[0]->{parentSpanId}, $parent_span_id,
     'Parent span of root should be the provided one');
+
+# span 0: root
 my $agent = $json_obj->{traces}->[0]->{spans}->[0]->{labels}->{'trace.cloud.google.com/agent'};
 is($agent, 'esp/' . ServiceControl::get_version(), 'Agent is set to "esp/xxx".');
 my $rootid = $json_obj->{traces}->[0]->{spans}->[0]->{spanId};
-is($json_obj->{traces}->[0]->{spans}->[1]->{name}, 'CheckServiceControl',
-    'Next trace span is CheckServiceControl');
+
+# span 1: FetchAccessToken
+is($json_obj->{traces}->[0]->{spans}->[1]->{name}, 'FetchAccessToken',
+    'Next trace span is FetchAccessToken');
 is($json_obj->{traces}->[0]->{spans}->[1]->{parentSpanId}, $rootid,
-    'Parent of CheckServiceControlCache span is root');
-my $check_service_control_id = $json_obj->{traces}->[0]->{spans}->[1]->{spanId};
-is($json_obj->{traces}->[0]->{spans}->[2]->{name}, 'CheckServiceControlCache',
+    'Parent of FetchAccessToken span is root');
+
+# span 2: CheckServiceControl
+is($json_obj->{traces}->[0]->{spans}->[2]->{name}, 'CheckServiceControl',
+    'Next trace span is CheckServiceControl');
+is($json_obj->{traces}->[0]->{spans}->[2]->{parentSpanId}, $rootid,
+    'Parent of CheckServiceControl span is root');
+my $check_service_control_id = $json_obj->{traces}->[0]->{spans}->[2]->{spanId};
+
```

Comment:
Hi Wayne, look like you are going to keep the `CheckServiceControl`.

Would be better to trigger all the original seven traces plus your added access token one?

Reply:
Yes, will do